pkg/file/pipeline/result_writer.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import "errors"
+
+type resultWriter struct {
+	target *pipeWriteArgs
+}
+
+func NewResultWriter(b *pipeWriteArgs) ChainWriter {
+	return &resultWriter{target: b}
+}
+
+func (w *resultWriter) ChainWrite(p *pipeWriteArgs) (int, error) {
+	*w.target = *p
```

Comment:
Is dereference needed?

Reply:
i believe so. otherwise you just set the address of the local pointer to be this of `p`, but i need the content to be visible in the calling context in hashTrieWriter

Reply:
I am not sure what the result writer purpose is. It just sets the target and I do not see that it it used anywhere. I am not sure how hashTrieWriter uses this information set here.

Reply:
hmm... it is true, i can already just mutate the fields in `pipeWriteArgs`. in hashTrieWriter it is used since the target pointer here is set when `pipelineFn` is called, then once we reach this line it dereferences the content onto the pointer provided in the previous call to create the short pipeline. is it ok that i do this change as part of the integration PR?

Reply:
Yes, it is, thanks.